homeassistant/components/sensor/gpsd.py

```diff
@@ -0,0 +1,94 @@
+"""
+Support for getting information from GPSD.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.gpsd/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.components.gpsd as gpsd
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import (CONF_PLATFORM, CONF_MONITORED_CONDITIONS)
+
+DEPENDENCIES = ['gpsd']
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_TYPES = {
+    'time': ['Time', None, 'time'],
+    'latitude': ['Latitude', 'N', 'lat'],
+    'longitude': ['Longitude', 'E', 'lon'],
+    'altitude': ['Altitude', 'ft', 'alt'],
+    'speed': ['Speed', 'mph', 'speed'],
+    'climb': ['Climb', 'mph', 'climb'],
+    'track': ['Track', 'miles', 'track'],
+    'fix': ['Fix', None, 'mode']
+}
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'gpsd',
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=[]):
+        [vol.In(SENSOR_TYPES.keys())],
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the GPSD platform."""
+    if gpsd.GPSD is None:
+        _LOGGER.error('A connection has not been made to the GPSD.')
+        return False
+
+    dev = []
+    try:
+        for condition in config['monitored_conditions']:
+            if condition not in SENSOR_TYPES:
+                _LOGGER.error('Sensor type: "%s" does not exist', condition)
+            else:
+                dev.append(GpsdSensor(condition))
+    except KeyError:
+        pass
+
+    add_devices(dev)
+
+
+class GpsdSensor(Entity):
+    """Representation of an GPSD Sensor."""
+
+    def __init__(self, condition):
+        """Initialize the sensor."""
+        self._name = SENSOR_TYPES[condition][0]
+        self._state = None
+        self._unit_of_measurement = SENSOR_TYPES[condition][1]
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def name(self):
+        """Get the name of the sensor."""
+        return self._name
+
+    def update(self):
+        """Get the latest value from GPSD."""
+        gps_data = gpsd.GPSD.get_data()
+
+        if self._name == 'Time':
```

Comment:
Wouldn't it make more sense if we have all this data as attributes on the sensor ? That way it would also show up on the map 👍  (we pick up all entities that have `ATTR_LATITUDE` and `ATTR_LONGITUDE` set)